packages/react-ui-core/src/Ratings/RatingBar.js

```diff
@@ -0,0 +1,37 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import RatingItem from './RatingItem'
+
+export default class RatingBar extends Component {
+  static propTypes= {
+    theme: PropTypes.object,
+  }
+  static defaultProps = {
+    theme: {},
+  }
+
+  render() {
+    const { max, rating, backgroundColor, total, theme } = this.props
+    const ratingItems = []
+    let currentRating = rating
+    while (ratingItems.length < max) {
+      const ratingValue = currentRating - 1 > 0 ?
+        1 : currentRating
+      currentRating -= 1
+      const val = ratingValue < 0 ? 0 : ratingValue
+      ratingItems.push(
+        <RatingItem
+          width={`${val * 100}%`}
+          index={currentRating}
+          backgroundColor={backgroundColor}
+          theme={theme}
+        />)
+    }
+    return (
+      <div className="rating-bar">
+        <div className={theme.ratingcomp}>{ratingItems}</div>
+        <div className={theme.label}>(<span>{total}</span> {'Ratings'})</div>
```

Comment:
It's not really necessary to use it. Using `{}` in jsx code tells the code to interpret what is inside.  In this case it's just a string so it's an extra step that jsx has to handle.

Please don't rely on ag.js to tell you how to code this.  You should be using ag.js as a guideline, not a definitive answer on how to do implement generic code.  I'm more concerned about enforcing text here in general which should *not* be enforced at all. It should be optional at best. There will be places on the ui where text will not be used and this component needs to be flexible enough to handle that scenario.